python/cudf/cudf/core/frame.py

```diff
@@ -1202,6 +1204,171 @@ def __arrow_array__(self, type=None):
             "consider using .to_arrow()"
         )
 
+    def _sample(
```

Comment:
Can we add an NVTX range to this function and then does it make sense to just expose this as `def sample` and inherit it in DataFrame / Series?